packages/backend/src/services/SpaceService/SpaceService.ts

```diff
@@ -155,12 +155,50 @@ export class SpaceService extends BaseService {
         ) {
             throw new ForbiddenError();
         }
+
+        // If this space has a parent, verify the user has access to the parent space
+        // 🙋 How recursive is this?
+        if (space.parentSpaceUuid) {
+            const parentSpace = await this.spaceModel.getSpaceSummary(
+                space.parentSpaceUuid,
+            );
+
+            // Ensure parent space is in the same project
+            if (parentSpace.projectUuid !== projectUuid) {
+                throw new ForbiddenError(
+                    'Parent space must be in the same project',
+                );
+            }
+
+            // Check if user has manage access to parent space
+            const parentSpaceAccess = await this.spaceModel.getUserSpaceAccess(
+                user.userUuid,
+                space.parentSpaceUuid,
+            );
```

Comment:
in here, I would call a `getRootSpace` so I could check the root space's permissions before going ahead